### PR TITLE
GitHub pipeline

### PR DESCRIPTION
**Pull request by MarcellvanRooyen**
Add github actions workflow to build and deploy mendix package

### CODE REVIEW

File: .github/workflows/Build_and_publish_package.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0 # avoid shallow clone so nbgv can do its work.
+    # Add tool nbgv for build support
+    - name: Update Gitversioning
+      run: |
+        dotnet tool install --tool-path . nbgv
+        .\nbgv cloud
```
Comment (ghost) [inline .github/workflows/Build_and_publish_package.yml]: Can we check if we could use their action instead? https://github.com/marketplace/actions/nerdbank-gitversioning

File: .gitignore [hunk 2]
```diff
@@ -36,6 +36,8 @@ _ReSharper*/
 *.userprefs
 *.swp
 *.DotSettings
+nuget.package/lib/
+nuget.pacakge/App_Readme/
```
Comment (ghost) [inline .gitignore]: typo!

File: LibGit2Sharp.Tests/GlobalSettingsFixture.cs [hunk 3]
```diff
@@ -53,33 +53,33 @@ public void TryingToResetNativeLibraryPathAfterLoadedThrows()
...
-            }
-        }
+        //        Assert.Empty(output);
+        //        Assert.Equal(0, exitCode);
+        //    }
+        //    finally
+        //    {
+        //        DirectoryHelper.DeleteDirectory(tempDir);
+        //    }
+        //}
```
Comment (ghost) [inline LibGit2Sharp.Tests/GlobalSettingsFixture.cs:83]: 🤔 If we want to address it later, let's create an issue in the repo so we don't forget about it.

File: LibGit2Sharp.sln.DotSettings [hunk 4]
```diff
@@ -11,6 +11,9 @@
 	<s:Boolean x:Key="/Default/CodeStyle/CodeFormatting/CSharpFormat/SPACE_WITHING_EMPTY_BRACES/@EntryValue">True</s:Boolean>
 	<s:Boolean x:Key="/Default/CodeStyle/CodeFormatting/CSharpFormat/SPACE_WITHIN_SINGLE_LINE_ARRAY_INITIALIZER_BRACES/@EntryValue">True</s:Boolean>
 	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/PredefinedNamingRules/=PrivateInstanceFields/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="" Suffix="" Style="aaBb" /&gt;</s:String>
+	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002ECSharp_002ECodeStyle_002ECSharpKeepExistingMigration/@EntryIndexedValue">True</s:Boolean>
+	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002ECSharp_002ECodeStyle_002ECSharpPlaceEmbeddedOnSameLineMigration/@EntryIndexedValue">True</s:Boolean>
+	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002ECSharp_002ECodeStyle_002ECSharpUseContinuousIndentInsideBracesMigration/@EntryIndexedValue">True</s:Boolean>
```
Comment (ghost) [inline LibGit2Sharp.sln.DotSettings:16]: Not sure if it's okay to keep it here?

File: LibGit2Sharp/LibGit2Sharp.csproj [hunk 5]
```diff
@@ -32,7 +32,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="LibGit2Sharp.NativeBinaries" Version="[2.0.305]" PrivateAssets="none" />
+    <PackageReference Include="LibGit2Sharp.NativeBinaries" Version="2.0.312" />
```
Comment (ghost) [inline LibGit2Sharp/LibGit2Sharp.csproj]: We probably need to reference our package instead. Also might require adding a proper NuGet package source.

File: nuget.package/App_Readme/CHANGES.md [hunk 6]
```diff
@@ -0,0 +1,782 @@
+# LibGit2Sharp Changes
```
Comment (A-Ovchinnikov-mx) [inline nuget.package/App_Readme/CHANGES.md:1]: We probably don't care about including this file and README.md into our package since it's intended for internal use only anyway. But we should keep the LICENSE.md.

File: nuget.package/Mendix_package.nuspec [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    <authors>LibGit2Sharp contributors</authors>
+    <owners>mendix</owners>
+    <licenseUrl>https://raw.githubusercontent.com/libgit2/libgit2/master/COPYING</licenseUrl>
+    <projectUrl>https://github.com/libgit2/libgit2sharp.nativebinaries</projectUrl>
+    <repository url="https://github.com/mendix/libgit2sharp.git" type="git"/>
+    <iconUrl>https://raw.githubusercontent.com/libgit2/libgit2sharp/master/square-logo.png</iconUrl>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>LibGit2Sharp brings all the might and speed of libgit2, a native Git implementation, to the managed world of .Net and Mono.</description>
+  </metadata>
+</package>
```
Comment (ghost) [inline nuget.package/Mendix_package.nuspec]: You probably need to specify input files here (including all the `*.md` files). Otherwise it doesn't really pack anything 😄 

We also wouldn't need the `App_ReadMe` folder then because you can specify this in `.nuspec`.

File: .github/workflows/Build_and_publish_package.yml (hunk 1 above)
Comment (MarcellvanRooyen) [reply .github/workflows/Build_and_publish_package.yml]: Changed to using the github action

File: .gitignore (hunk 2 above)
Comment (MarcellvanRooyen) [reply .gitignore]: fixed

File: LibGit2Sharp.Tests/GlobalSettingsFixture.cs (hunk 3 above)
Comment (MarcellvanRooyen) [reply LibGit2Sharp.Tests/GlobalSettingsFixture.cs:83]: https://github.com/mendix/libgit2sharp/issues/2#issue-702650660

issue created

File: LibGit2Sharp.sln.DotSettings (hunk 4 above)
Comment (MarcellvanRooyen) [reply LibGit2Sharp.sln.DotSettings:16]: Will probably just get re-added by resharper again at some point but i have removed them for now

File: LibGit2Sharp/LibGit2Sharp.csproj (hunk 5 above)
Comment (MarcellvanRooyen) [reply LibGit2Sharp/LibGit2Sharp.csproj]: Updated the reference to Mendix.LibGit2Sharp.NativeBinaries version 1.101.1

File: nuget.package/Mendix_package.nuspec (hunk 7 above)
Comment (MarcellvanRooyen) [reply nuget.package/Mendix_package.nuspec]: I have retested this the package works without having to specify any files. I double checked the package and it has the correct contents just had to copy the files correctly as stated

File: LibGit2Sharp/LibGit2Sharp.csproj (hunk 5 above)
Comment (MarcellvanRooyen) [reply LibGit2Sharp/LibGit2Sharp.csproj]: increased the build no to 1.101.4 to test the package.
i will revert this once github clears the packages

File: LibGit2Sharp/LibGit2Sharp.csproj [hunk 8]
```diff
@@ -18,6 +18,7 @@
     <AssemblyOriginatorKeyFile>..\libgit2sharp.snk</AssemblyOriginatorKeyFile>
     <PackageIcon>square-logo.png</PackageIcon>
     <PackageLicenseFile>App_Readme/LICENSE.md</PackageLicenseFile>
+    <UserSecretsId>cdd90144-8692-4d6d-bb35-66e4947ab673</UserSecretsId>
```
Comment (ghost) [inline LibGit2Sharp/LibGit2Sharp.csproj]: Why is this required?

File: .github/workflows/Build_and_publish_package.yml [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+      shell: powershell #Keep as powershell and not pwsh
+    # Build and deploy nuget package
+    # We include the GitHub package source so that we can reference current packages and also so that we can push to the package repo
+    # After build we copy the binaries to package output folder
+    # Finally we pack and push the package to github packages        
+    - name: Build & Deploy Package
+      run: |
+        nuget source Add -Name "GitHub" -Source "https://nuget.pkg.github.com/mendix/index.json" -UserName MarcellVanRooyen -Password ${{ secrets.GHPACKAGESTOKEN }}
+        dotnet build -c Release
+        copy .\bin\LibGit2Sharp\Release\net46\*.* .\nuget.package\lib\net46\ -Force
```
Comment (ghost) [inline .github/workflows/Build_and_publish_package.yml]: Consider specifying these files in .nuspec, or creating a script to collect them and build the package (instead of doing this here).

File: .gitignore [hunk 10]
```diff
@@ -36,6 +36,8 @@ _ReSharper*/
 *.userprefs
 *.swp
 *.DotSettings
+nuget.package/lib/
```
Comment (ghost) [inline .gitignore]: You don't use `lib` there anymore, so it can go from .gitignore. And if you specify the LICENSE.md in the .nuspec, then you can also let go the App_Readme folder (here and on the disk).

**Review by ghost — CHANGES_REQUESTED** [review]

File: LibGit2Sharp/LibGit2Sharp.csproj (hunk 8 above)
Comment (MarcellvanRooyen) [reply LibGit2Sharp/LibGit2Sharp.csproj]: its not I removed it

File: .gitignore (hunk 10 above)
Comment (MarcellvanRooyen) [reply .gitignore]: updated accordingly

File: .github/workflows/Build_and_publish_package.yml (hunk 9 above)
Comment (MarcellvanRooyen) [reply .github/workflows/Build_and_publish_package.yml]: moved file references to .nuspec file and removed them from the yml file

File: LibGit2Sharp/LibGit2Sharp.csproj [hunk 11]
```diff
@@ -32,7 +32,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="LibGit2Sharp.NativeBinaries" Version="[2.0.305]" PrivateAssets="none" />
+    <PackageReference Include="Mendix.LibGit2Sharp.NativeBinaries" Version="1.101.1" />
```
Comment (ghost) [inline LibGit2Sharp/LibGit2Sharp.csproj]: Now we need 1.101.10

**Review by ghost — APPROVED** [review]
